Source/santa-driver/SantaDecisionManager.cc

```diff
@@ -316,10 +320,16 @@ santa_action_t SantaDecisionManager::GetFromDaemon(
       return ACTION_ERROR;
     }
 
+    // Check the cache every kRequestLoopSleepMilliseconds. Break this loop and send the request
+    // again if kRequestCacheChecks is reached. Don't break the loop if the daemon is working on the
+    // request, indicated with ACTION_RESPOND_ACK.
+    auto cache_check_count = 0;
     do {
       msleep((void *)message->vnode_id, NULL, 0, "", &ts_);
       return_action = GetFromCache(identifier);
-    } while (return_action == ACTION_REQUEST_BINARY && ClientConnected());
+      if (return_action == ACTION_REQUEST_BINARY) ++cache_check_count;
+    } while ((return_action == ACTION_REQUEST_BINARY || return_action == ACTION_RESPOND_ACK)
+             && ClientConnected() && cache_check_count < kRequestCacheChecks);
```

Comment:
I think the condition here can include the cache_check_count increment too without being less clear. Maybe this:

```c++
} while (ClientConnected() &&
         ((return_action == ACTION_REQUEST_BINARY && ++cache_check_count < kRequestCacheChecks) ||
          (return_action == ACTION_RESPOND_ACK)));
```

Reply:
Yeah, that's nicer. Except the || is over 100 lines :(